client/txn.go

```diff
@@ -77,30 +85,200 @@ type txn struct {
 	haveEndTxn   bool // True if there was an explicit EndTransaction
 }
 
-func (t *txn) init(kv *kv) {
-	t.kv = *kv
-	t.wrapped = kv.Sender
-	t.txnSender.txn = t
-	t.kv.Sender = &t.txnSender
+func newTx(db *DB, depth int) *Tx {
+	tx := &Tx{
+		db:      *db,
+		wrapped: db.Sender,
+	}
+	tx.txnSender.Tx = tx
+	tx.db.Sender = &tx.txnSender
```

Comment:
if the db's Sender is equal to tx's txnsender, why have both? Tx could have a getter that fetches from db, no?